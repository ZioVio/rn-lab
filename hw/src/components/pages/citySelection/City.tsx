import React from 'react';
import { StyleSheet, View } from 'react-native';
import { getReadableStateName } from '../../../enums/weatherState';
import ShortCity from '../../../models/shortCity';
import { AppText, AppTextSecondary } from '../../shared/Text';
import useNumberToTemperatureTextConverter from '../../../hooks/useNumberToTemperatureTextConverter';
import BorderBottom from '../../shared/BorderBottom';

interface CityProps {
  city: ShortCity;
}

export default function City({ city }: CityProps) {
  const weatherState: string = getReadableStateName(city.weatherState);
  const temperature: string = useNumberToTemperatureTextConverter(city.temperature);
  const weatherStateString: string = weatherState && ('// ' + weatherState.toUpperCase());
  return (
    <BorderBottom style={styles.container}>
      <View style={styles.temperature}>
        <AppText size={60} testID="City.Temperature">{temperature}</AppText>
      </View>
      <View style={styles.details}>
        <AppText>{city.name}</AppText>
        <AppTextSecondary size={20} testID="City.WeatherState" >{weatherStateString}</AppTextSecondary>
      </View>
    </BorderBottom>
  );
}


const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 15,
  },
  temperature: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 3,
  },
  details: {
    justifyContent: 'center',
    flex: 4,
  },
});
