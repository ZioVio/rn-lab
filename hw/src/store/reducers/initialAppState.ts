import AppState from '../../models/store/appState';
import { initialState as initialAuthState } from './authReducer';
import { initialState as initialCitySelectionState } from './citySelectionReducer';
import { initialState as initialDaySelectionState } from './daySelectionReducer';
import { initialState as initialSettingsState } from './settingsReducer';
import { initialState as initialSharedState } from './sharedReducer';
import { initialState as initialWeatherDetails } from './weatherDetailsReducer';
import { initialState as initialWeatherSearchState } from './weatherSearchReducer';

const initialAppState: AppState = {
  auth: initialAuthState,
  citySelection: initialCitySelectionState,
  daySelection: initialDaySelectionState,
  settings: initialSettingsState,
  shared: initialSharedState,
  weatherDetails: initialWeatherDetails,
  weatherSearch: initialWeatherSearchState,
};

export default initialAppState;
