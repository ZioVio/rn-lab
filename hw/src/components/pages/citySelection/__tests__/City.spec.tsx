import React from 'react';
import City from '../City';
import { render } from '@testing-library/react-native';
import ShortCity from '../../../../models/shortCity';
import WeatherState from '../../../../enums/weatherState';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import initialAppState from '../../../../store/reducers/initialAppState';

const mockStore = configureStore();
const createMockedStore = () => mockStore(initialAppState);

const city: ShortCity = {
  id: '2131',
  name: 'something',
  temperature: 234,
  weatherState: WeatherState.CLOUDY,
};

test('should render as expected', () => {
  const { toJSON } = render(
    <Provider store={createMockedStore()}>
      <City city={city} />
    </Provider>
  );

  expect(toJSON()).toMatchSnapshot();
});

test('temperature is displayed', () => {
  const { getByTestId } = render(
    <Provider store={createMockedStore()}>
      <City city={city} />
    </Provider>
  );

  expect(() => getByTestId('City.Temperature')).not.toThrowError();
  expect(getByTestId('City.Temperature')?.children[0]?.toString()).toBeTruthy();
});

test('city name is displayed', () => {
  const { getByText } = render(
    <Provider store={createMockedStore()}>
      <City city={city} />
    </Provider>
  );

  expect(getByText(city.name)).not.toBeNull();
});

test('weather state name is displayed', () => {
  const { getByTestId } = render(
    <Provider store={createMockedStore()}>
      <City city={city} />
    </Provider>
  );

  expect(() => getByTestId('City.WeatherState')).not.toThrowError();
  expect(getByTestId('City.WeatherState')?.children[0]?.toString()).toBeTruthy();
});


