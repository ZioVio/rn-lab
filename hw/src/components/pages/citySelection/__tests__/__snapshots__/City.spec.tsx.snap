// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#45464f",
        "borderBottomWidth": 2,
      },
      Object {
        "flexDirection": "row",
        "paddingVertical": 15,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <Text
      size={60}
      style={
        Array [
          Object {
            "color": "#fffffe",
            "fontFamily": "PTS55F",
          },
          Object {
            "fontSize": 22,
          },
          Object {
            "fontSize": 60,
          },
          undefined,
        ]
      }
      testID="City.Temperature"
    >
      -39Â°C
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 4,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fffffe",
            "fontFamily": "PTS55F",
          },
          Object {
            "fontSize": 22,
          },
          undefined,
          undefined,
        ]
      }
    >
      something
    </Text>
    <Text
      size={20}
      style={
        Array [
          Object {
            "color": "#fffffe",
            "fontFamily": "PTS55F",
          },
          Object {
            "fontSize": 22,
          },
          Object {
            "fontSize": 20,
          },
          Object {
            "color": "#45464f",
          },
        ]
      }
      testID="City.WeatherState"
    >
      // CLOUDY
    </Text>
  </View>
</View>
`;
